plugins/processors/scale/scale_test.go

```diff
@@ -0,0 +1,283 @@
+package Scale
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestScaler(t *testing.T) {
+	tests := []struct {
+		name     string
+		scale    *Scale
+		inputs   []telegraf.Metric
+		expected []telegraf.Metric
+	}{
+		{
+			name: "Field Scaling",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  -1,
+						InMax:  1,
+						OutMin: 0,
+						OutMax: 100,
+						Fields: []string{"test1", "test2"},
+					},
+					{
+						InMin:  -5,
+						InMax:  0,
+						OutMin: 1,
+						OutMax: 9,
+						Fields: []string{"test3", "test4"},
+					},
+				},
+			},
+			inputs: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": int64(0),
+						"test2": uint64(1),
+					}, time.Unix(0, 0)),
```

Comment:
For simplifying the test definitions, you could also use `TestMetric`. Example:
```suggestion
				testutil.TestMetric(int64(0), "test1"),
				testutil.TestMetric(uint64(1), "test2"),
```

Reply:
You could adapt your filters so they match field "value", isn't it?

Reply:
Yes, but in that case I can't have test cases where metrics define multiple fields and only a subset of them are scaled. If it is ok not to have this test case, sure. I can do that.

Reply:
You could add this one specific case in a separate test `testMultipleFieldSelect` or something?